src/block/WallHangingSign.php

```diff
@@ -50,16 +51,38 @@ protected function recalculateCollisionBoxes() : array{
 	}
 
 	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
-		if(Facing::axis($face) === Axis::Y){
+		if($player === null){
 			return false;
 		}
+		$attachFace = Facing::axis($face) === Axis::Y ? Facing::rotateY($player->getHorizontalFacing(), clockwise: true) : $face;
+		$direction = null;
 
-		$this->facing = Facing::rotateY($face, clockwise: true);
+		foreach([
+			$attachFace,
+			Facing::opposite($attachFace)
+		] as $side){
+			if($this->canBeSupportedAt($blockReplace->getSide($side), $side)){
+				$direction = $side;
+				break;
+			}
+		}
```

Comment:
I feel like a simple if with 2 canBeSupportedAt calls would be less code...

Reply:
This code is now really confusing. If the player clicked a horizontal face, this extra logic shouldn't apply